src/main/java/edu/kis/powp/jobs2d/features/DrawerPanelClickMouseListenerFeature.java

```diff
@@ -10,36 +10,28 @@
 
 public class DrawerPanelClickMouseListenerFeature implements MouseListener {
     private final Application application;
-    private ILine line;
     private int startX = 0, startY = 0;
-    private String name;
-    private boolean isClicked = false;
+    private boolean isClicked = true;
     private DriverManager driverManager;
 
-    private DrawPanelController drawController;
 
-    public DrawerPanelClickMouseListenerFeature(DrawPanelController drawController,
-                                                Application application, DriverManager driverManager, ILine line, String name) {
+    public DrawerPanelClickMouseListenerFeature(Application application, DriverManager driverManager) {
         super();
-        this.drawController = drawController;
-        this.line = line;
         this.application = application;
         this.driverManager = driverManager;
-        this.name = name;
     }
 
     @Override
     public void mouseClicked(MouseEvent mouseEvent) {
         startX = mouseEvent.getX() - (application.getFreePanel().getSize().width/2);
```

Comment:
Unnecessary coupling to Application. It should be enough to keep freePanel reference.